### ISSUE TITLE
FYI: Syntax deprecation rules do not check data source or output blocks (Upstream from Terraform)

### ISSUE THREAD

**Issue by gdavison** [terraform-linters/tflint#872]
The internal syntax checking for Terraform does not check syntax for data source attributes or output block `value` fields. This carries over to `tflint` checks.

Reported in https://github.com/hashicorp/terraform/issues/25828.

For example, in the following configuration

```terraform
variable "api_gateway_name" {
  type = string
}

data "aws_api_gateway_rest_api" "selected" {
  name = "${var.api_gateway_name}"
}

data "aws_cognito_user_pools" "selected" {
  name = "selected"
}

resource "aws_api_gateway_authorizer" "cognito" {
  name          = "cognito"
  type          = "COGNITO_USER_POOLS"
  rest_api_id   = "${data.aws_api_gateway_rest_api.selected.id}"
  provider_arns = data.aws_cognito_user_pools.selected.arns
}

output "api_id" {
  value = "${aws_api_gateway_authorizer.cognito.rest_api_id}"
}
```

### Expected

Issues should be reported for
* data source `aws_api_gateway_rest_api.selected` `name`
* resource `aws_api_gateway_authorizer.cognito` `rest_api_id`
* output `api_id` `value`

### Actual

An issue is reported for 
* resource `aws_api_gateway_authorizer.cognito` `rest_api_id`

## Version

TFLint v0.18.0
Terraform v0.12.29 and 0.13.0

**Comment by bendrucker** [comment]
I think these are similar issues with different causes. I _think_ this one might be fixed already by #849. That hasn't been released yet.

**Comment by wata727** [comment]
Yes, the fix has been released in v0.19.0. Please try it again. Thanks.
